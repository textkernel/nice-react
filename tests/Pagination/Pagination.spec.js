import React from 'react';
import { Pagination } from '../../dist';


describe('<Pagination />', () => {
    it('should render default Pagination', () => {
        const wrapper = shallow(<Pagination>
            <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
            <Pagination.Item current>1</Pagination.Item>
            <Pagination.Item>2</Pagination.Item>
            <Pagination.Item>3</Pagination.Item>
            <Pagination.Gap />
            <Pagination.Item>4</Pagination.Item>
            <Pagination.Item>5</Pagination.Item>
            <Pagination.Item>Next &raquo;</Pagination.Item>
        </Pagination>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render small Pagination', () => {
        const wrapper = shallow(<Pagination size="small">
            <Pagination.Item current>1</Pagination.Item>
            <Pagination.Item>2</Pagination.Item>
            <Pagination.Gap />
            <Pagination.Item>4</Pagination.Item>
            <Pagination.Item>5</Pagination.Item>
        </Pagination>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render large Pagination', () => {
        const wrapper = shallow(<Pagination size="large">
            <Pagination.Item current>1</Pagination.Item>
            <Pagination.Item>2</Pagination.Item>
            <Pagination.Gap />
            <Pagination.Item>4</Pagination.Item>
            <Pagination.Item>5</Pagination.Item>
        </Pagination>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render bordered Pagination', () => {
        const wrapper = shallow(<Pagination bordered>
            <Pagination.Item current>1</Pagination.Item>
            <Pagination.Item>2</Pagination.Item>
            <Pagination.Gap />
            <Pagination.Item>4</Pagination.Item>
            <Pagination.Item>5</Pagination.Item>
        </Pagination>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Pagination with custom className/style', () => {
        const wrapper = shallow(
            <Pagination className="my-custom-class" style={{ top: 0 }}>
                <Pagination.Item current>1</Pagination.Item>
            </Pagination>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
