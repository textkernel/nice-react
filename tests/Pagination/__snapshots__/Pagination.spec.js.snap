// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render bordered Pagination 1`] = `
<nav>
  <ul
    className="nice-pagination bordered"
    role="navigation"
  >
    <Item
      current={true}
      disabled={false}
      onClick={null}
    >
      1
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      2
    </Item>
    <Gap />
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      4
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      5
    </Item>
  </ul>
</nav>
`;

exports[`<Pagination /> should render default Pagination 1`] = `
<nav>
  <ul
    className="nice-pagination"
    role="navigation"
  >
    <Item
      current={false}
      disabled={true}
      onClick={null}
    >
      « Previous
    </Item>
    <Item
      current={true}
      disabled={false}
      onClick={null}
    >
      1
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      2
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      3
    </Item>
    <Gap />
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      4
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      5
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      Next »
    </Item>
  </ul>
</nav>
`;

exports[`<Pagination /> should render large Pagination 1`] = `
<nav>
  <ul
    className="nice-pagination large"
    role="navigation"
  >
    <Item
      current={true}
      disabled={false}
      onClick={null}
    >
      1
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      2
    </Item>
    <Gap />
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      4
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      5
    </Item>
  </ul>
</nav>
`;

exports[`<Pagination /> should render small Pagination 1`] = `
<nav>
  <ul
    className="nice-pagination small"
    role="navigation"
  >
    <Item
      current={true}
      disabled={false}
      onClick={null}
    >
      1
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      2
    </Item>
    <Gap />
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      4
    </Item>
    <Item
      current={false}
      disabled={false}
      onClick={null}
    >
      5
    </Item>
  </ul>
</nav>
`;
